src/rpdk/core/generate_stack_templates.py

```diff
@@ -0,0 +1,234 @@
+        return self.stack_template_config["target_folder_path"]
+
+    @property
+    def stack_template_dependency_file_name(self):
+        return self.stack_template_config["dependency_file_name"]
+
+    def generate_stack_templates(self) -> None:
+        """
+        Generate stack_template files based on the contract test input files.
+
+        This method checks if file generation is enabled and if the target contract test folder exists.
+        If both conditions are met, it creates the stack_template folder, copies the contract test dependencies,
+        and generates stack_template files for each contract test input file up to the specified count.
+        """
+        if not self.file_generation_enabled or not self.contract_test_folder_exists():
+            return
+        self._setup_stack_template_environment()
+        self._generate_stack_template_files()
+
+    def contract_test_folder_exists(self) -> bool:
+        return Path(self.target_contract_test_folder_path).exists()
+
+    def _setup_stack_template_environment(self) -> None:
+        stack_template_root = Path(self.stack_template_root_folder_path)
+        stack_template_folder = Path(self.target_stack_template_folder_path)
+        self.clean_and_create_stack_template_folder(
+            stack_template_root, stack_template_folder
+        )
+        self.create_stack_template_bootstrap(
+            Path(self.target_contract_test_folder_path), stack_template_root
+        )
+
+    def _generate_stack_template_files(self) -> None:
+        resource_name = self.type_name
+        stack_template_folder = Path(self.target_stack_template_folder_path)
+        contract_test_files = self._get_sorted_contract_test_files()
+        for count, ct_file in enumerate(contract_test_files, start=1):
+            self.create_stack_template_file(
+                resource_name, ct_file, stack_template_folder, self.file_prefix, count
+            )
+
+    def _get_sorted_contract_test_files(self) -> list:
+        contract_test_folder = Path(self.target_contract_test_folder_path)
+        contract_test_files = [
+            file
+            for file in contract_test_folder.glob(CONTRACT_TEST_INPUT_PREFIX)
+            if file.is_file() and file.name in self.contract_test_file_names
+        ]
+        return sorted(contract_test_files)
+
+    def clean_and_create_stack_template_folder(
+        self, stack_template: Path, stack_template_folder: Path
+    ) -> None:
+        """
+        Clean and create the stack_template folder.
+
+        This method removes the existing stack_template root folder and creates a new stack_template folder.
+
+        Args:
+            stack_template (Path): The path to the stack_template root folder.
+            stack_template_folder (Path): The path to the stack_template folder.
+        """
+        # shutil.rmtree(stack_template, ignore_errors=True)
+        stack_template_folder.mkdir(parents=True, exist_ok=True)
```

Comment:
does it really need to be a separate method? why 116 is commented?

Reply:
Removed commented method . it is not required